.circleci/main-config.yml

```diff
@@ -1079,35 +1095,15 @@ parameters:
 workflows:
   version: 2
 
-  # Run all tests on DSS project
-  dss-build-test-workflow:
+  # Triggered from a local script to test PR changes
+  combined-build-test-workflow:
     when:
-      and:
-        - equal: ["build-test", << pipeline.parameters.action >>]
-        - << pipeline.parameters.dss >>
+      equal: ["build-test", << pipeline.parameters.action >>]
     jobs:
       - dss-compile-and-run-test-suite-job
       - dss-parallel-compile-and-test-job:
-          parallelism: *test_parallelism
-
-  # Run tests on DSM
-  dsm-build-test-workflow:
-    when:
-      and:
-        - equal: ["build-test", << pipeline.parameters.action >>]
-        - << pipeline.parameters.dsm >>
-    jobs:
+          parallelism: << pipeline.parameters.test_parallelism >> # the caller of this workflow will tell us the parallelism
```

Comment:
This is an optimization `test_parallelism` pipeline parameter is by default 1.
Expectation is that if setup workflow (in config.yml file) does indeed want to run dss tests, then it will also set the parallelism level to something high (15 in our case).
But if there is no need to run dss tests, then we start the test job on the 1 instance and skip all the steps.